src/main/java/com/google/api/codegen/metacode/FieldStructureParser.java

```diff
@@ -65,36 +64,38 @@ private static InitValueConfig createInitValueConfig(
   }
 
   // Parses `path` to construct the `InitCodeNode` it specifies. `path` must be a valid config
-  // satisfying the recursive grammar below:
-  // config:
-  //   ident
-  //   config '.' ident
-  //   config '[' number ']'
-  //   config '{' number '}'
-  //   config '{' string '}'
-  //   config '{' ident '}' (for compatibility, the identifier is just treated as a string)
-  private static InitCodeNode parsePartialDottedPathToInitCodeNode(
-      String path, InitValueConfig initValueConfig) {
-    Scanner scanner = new Scanner(path);
+  // satisfying the eBNF grammar below:
+  //
+  // config = path ['=' value];
+  // path = ident pathElem* ['%' ident]
+  // pathElem = ('.' ident) | ('[' int ']') | ('{' value '}');
+  // value = int | string | ident;
+  //
+  // For compatibility with the previous parser, when ident is used as a value, the value is
+  // the name of the ident.
+  private static InitCodeNode parseConfig(
```

Comment:
Do we have a pre-existing unit test for this that continues to pass with your changes?

I presume that since baselines aren't changed, the uses that we capture remain unaffected....but it would be good to capture the intended behavior in a unit test.

Reply:
[`SampleInitCodeTest`](https://github.com/googleapis/gapic-generator/blob/master/src/test/java/com/google/api/codegen/metacode/SampleInitCodeTest.java) tests this. I made an intentional bug and verified that the test fails :)

Reply:
:+1: